std/experimental/allocator/building_blocks/fallback_allocator.d

```diff
@@ -24,8 +24,9 @@ struct FallbackAllocator(Primary, Fallback)
     import std.traits : hasMember;
     import std.typecons : Ternary;
 
-    // Need at least one stateless allocator for this to work
-    static if (!stateSize!Primary || !stateSize!Fallback)
+    // Need both allocators to be stateless
+    // This is to avoid using default initialized stateful allocators
+    static if (!stateSize!Primary && !stateSize!Fallback)
```

Comment:
Clever idea. I suspect it won't run tests on some stateful allocators that would work properly, such as `InSituRegion`.

Can you try being a bit more specific? For example:

```
@system unittest
{
    Primary testing;
    if (testing.allocate(1) is null)
        return; // cannot test with degenerate primary allocator
    testAllocator!(() => FallbackAllocator());
}
```

Reply:
Thanks for the comment, fixing

Reply:
However, would this be a strong enough guarantee that the allocator is or isn't degenerate?
``testing.allocate`` in the case mentioned above could return not null and the allocator still be ill formed.

I think we should manually add tests for for stateful allocators used as parameters for the FallbackAllocator.